delphi/app/pki_model.py

```diff
@@ -0,0 +1,106 @@
+from delphi.app.graph import run_optimization
+
+
+class PKI:
+    """
+    PKI is a class to represent the building PKI.
+
+    Attributes:
+        vertices: a dictionary of nodes and their enabled status
+        l2vertices: a dictionary of nodes and their enabled status
+        l3vertices: a dictionary of nodes and their enabled status
+        s1: the people count of room s1
+        s2: the people count of room s2
+        s3: the people count of room s3
+        desired_flow: the amount of people that need to exit pki
+        edges: a dictionary of edges and their weights
+        l2edges: a dictionary of edges and their weights
+        l3edges: a dictionary of edges and their weights
+    """
+
+    def __init__(self):
+        """
+        PKI constructor
+        """
+        self.vertices = {'t1': True, 't2': True, 's1': True, 's2': True, 's3': True, 'h1': True, 'h2': True, 'h5': True}
+        self.l2vertices = {'t3': True, 't4': True, 'h3': True, 'h4': True}
+        self.l3vertices = {'h6': True}
```

Comment:
maybe move the vertices dicts into a list of dicts named 'levels'. assuming l2 and l3 refer to the floor level the vertices belong to.

should also create a setVertices method, similar to what was done for edges for uniformity

Reply:
So the reason there are 3 different dictionaries for both vertices and edges is because of the way we are optimizing. The idea is that we will try to optimize with a small graph and then make the graph bigger if proper routes cannot be found. There are three sizes of graphs, vertices, l2vertices, and l3vertices. We could mash these in to one dictionary, but I figured 3 would be better for readability. As for making a set vertices method, I'll go ahead and do that!